src/main/java/com/ericsson/ei/frontend/BackEndInformationController.java

```diff
@@ -16,35 +16,112 @@
+        try {
+            String nameOfDeletedInstance = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            backEndInstancesUtils.setInstances(new JsonParser().parse(nameOfDeletedInstance).getAsJsonArray());
+            backEndInstancesUtils.writeIntoFile();
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", getHeaders(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
-    @CrossOrigin
     @RequestMapping(value = "/add-instances", method = RequestMethod.POST)
-    ResponseEntity<String> addInstanceInformation(Model model, HttpServletRequest request);
+    public ResponseEntity<String> addInstanceInformation(Model model, HttpServletRequest request) {
+        try {
+            String nameOfNewInstance = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            JsonObject instance = new JsonParser().parse(nameOfNewInstance).getAsJsonObject();
+            if (!backEndInstancesUtils.checkIfInstanceAlreadyExist(instance)) {
+                backEndInstancesUtils.getInstances().add(instance);
+                backEndInstancesUtils.writeIntoFile();
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>("Instance already exist", getHeaders(), HttpStatus.BAD_REQUEST);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", getHeaders(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value = "/switchBackend", method = RequestMethod.POST)
+    public ResponseEntity<String> switchBackEndInstanceByMainPage(Model model, HttpServletRequest request) {
+        try {
+            String backEndName = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            List<BackEndInformation> info = new ArrayList<>();
+            for (BackEndInformation backEndInformation : backEndInstancesUtils.getInformation()) {
+                backEndInformation.setActive(false);
+                if (backEndInformation.getName().equals(backEndName)) {
+                    backEndInstancesUtils.setBackEndProperties(backEndInformation);
+                    backEndInformation.setActive(true);
+                }
+                info.add(backEndInformation);
+            }
+            backEndInstancesUtils.setInformation(info);
+            JsonArray result = (JsonArray) new Gson().toJsonTree(backEndInstancesUtils.getInformation(), new TypeToken<List<BackEndInformation>>() {
+            }.getType());
+            backEndInstancesUtils.setInstances(result);
+            backEndInstancesUtils.writeIntoFile();
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", getHeaders(), HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Include Exception message into the response message. Include more info of what failed(or what possibly failed.)
"Internal error" doesn't tell user/admin that much.

Reply:
getHeaders(),, should be added to all Entrypoints that return Responses where JS Ajax requests expects JSON as ContentType Header.

Reply:
Check the JS code and Ajax request and what the Ajax request expects from these entry points.

Reply:
added